samples/bluetooth/broadcaster_multiple/src/broadcaster_multiple.c

```diff
@@ -78,20 +82,56 @@ int broadcaster_multiple(void)
 	int err;
 
 	for (int index = 0; index < CONFIG_BT_EXT_ADV_MAX_ADV_SET; index++) {
+		const struct bt_data *ad;
+		size_t ad_size;
+
 		/* Use advertising set instance index as SID */
 		adv_param.sid = index;
 
+		/* Use Coded PHY */
+		if ((index % 4) == 3) {
+			adv_param.options = BT_LE_ADV_OPT_EXT_ADV;
+			adv_param.options |= BT_LE_ADV_OPT_CODED;
+			ad = ad_long;
+			ad_size = ARRAY_SIZE(ad_long);
+		/* Use 1M auxiliary PDU */
+		} else if ((index % 4) == 2) {
+			adv_param.options = BT_LE_ADV_OPT_EXT_ADV;
+			adv_param.options |= BT_LE_ADV_OPT_NO_2M;
+			ad = ad_long;
+			ad_size = ARRAY_SIZE(ad_long);
+		/* Use 2M auxiliary PDU */
+		} else if ((index % 4) == 1) {
+			adv_param.options = BT_LE_ADV_OPT_EXT_ADV;
+			ad = ad_long;
+			ad_size = ARRAY_SIZE(ad_long);
+		/* Use 1M legacy PDU */
+		} else {
+			adv_param.options = BT_LE_ADV_OPT_NONE;
+			ad = ad_short;
+			ad_size = ARRAY_SIZE(ad_short);
+		}
```

Comment:
Consider moving this into a `select` function to simplify the loop. 

Also consider storing this in an array like

```c
struct {
    uint32_t options;
    const struct bt_data *ad;
    size_t ad_size
} adv_params[4];
```

So you can do things like `(index % sizeof(adv_params) == 2U)`. Alternatively you can also make it a circular linked list and just get the next entry for each iteration.